pkg/bpf/perf_linux.go

```diff
@@ -637,10 +637,19 @@ func NewPerCpuEvents(config *PerfEventConfig) (*PerCpuEvents, error) {
 		}
 	}
 
+	uba := bpfAttrMapOpElem{
+		mapFd: uint32(e.eventMap.fd),
+		flags: uint64(0),
+	}
+	ubaPtr := uintptr(unsafe.Pointer(&uba))
+	ubaSizeOf := unsafe.Sizeof(uba)
+
 	for _, event := range e.event {
 		// FIXME: Not sure what to do here, the map has already been updated and we can't
 		// fully restore it.
-		if err := e.eventMap.Update(event); err != nil {
+		uba.key = uint64(uintptr(unsafe.Pointer(&event.cpu)))
+		uba.value = uint64(uintptr(unsafe.Pointer(&event.Fd)))
+		if err := e.eventMap.Update(e.eventMap.fd, ubaPtr, ubaSizeOf); err != nil {
```

Comment:
If this is still being updated for each iteration through `e.event`, then I don't see why it's necessary to push this complexity up to `NewPerCpuEvents()`.

For that matter, this seems like it is not going to be called very frequently so the performance is not so critical in this path. I think it's good to reuse `UpdateElementWithUBA()` but this attribute initialization stuff could likely just be held in `Update()`.

Reply:
@joestringer yeah once I was done with this file I realized the same thing. I thought this was, somehow, called on every monitor call.